File: com.vyapaar/vyapaarservice/src/main/java/com/vyapaar/vyapaarservice/domain/Bill.java
package com.vyapaar.vyapaarservice.domain;


import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.mongodb.core.mapping.MongoId;
import java.util.List;

@AllArgsConstructor
@Data
public class Bill {
    @MongoId
    private String billId;
    private String customerName;
    private String customerPhoneNo;
    private String customerEmailID;
    private String customerAddress;
    private String billDate;
    private double totalBill;
    private double totalBillBeforeDiscount;
    private String payMode;
    private double discount;
    List<Product> purchasedProductList;
}

File: com.vyapaar/FrontEndService/vyapaarApp/src/model/bill.ts
import { PRODUCT } from "./product";

export type BILL =
{
  billId:string;
  customerName?:string;
  customerPhoneNo?:string;
  customerEmailID?:string;
  customerAddress?:string;
  billDate?:string;
  totalBill?:number;
  totalBillBeforeDiscount?:number;
  discount?:boolean;
  payMode?:string;
  purchasedProductList?:PRODUCT[];
}
